app/helpers/vita_partner_helper.rb

```diff
@@ -1,6 +1,6 @@
 module VitaPartnerHelper
   def grouped_organization_options
-    VitaPartner.top_level.collect do |partner|
+    VitaPartner.accessible_by(Ability.new(current_user)).top_level.collect do |partner|
       [partner.name, [[partner.name, partner.id], *partner.sub_organizations.collect { |v| [v.name, v.id] }]]
     end
   end
```

Comment:
In a controller, CanCanCan can preload the correct organizations that a user is allowed to access and make it available in the controller as `@organizations` or `@vita_partners`